From eaf97fe895e481615d106269e66d4fd6d22ac012 Mon Sep 17 00:00:00 2001
From: Etienne Fontaine <etienne.fontaine@blablacar.com>
Date: Wed, 2 Mar 2022 22:24:00 +0100
Subject: Remove default caBundle value from Chart to allow GitOps (#37655)

* Remove default caBundle value from Chart to allow GitOps

* Remove default caBundle value from Chart to allow GitOps

* Remove default caBundle value from Chart to allow GitOps
---
 manifests/charts/default/templates/mutatingwebhook.yaml   | 1 -
 manifests/charts/default/templates/validatingwebhook.yaml | 1 -
 .../istio-control/istio-discovery/files/gen-istio.yaml    | 4 ----
 .../istio-discovery/templates/mutatingwebhook.yaml        | 1 -
 .../istio-discovery/templates/revision-tags.yaml          | 1 -
 .../templates/validatingwebhookconfiguration.yaml         | 1 -
 .../charts/istiod-remote/templates/mutatingwebhook.yaml   | 1 -
 .../templates/validatingwebhookconfiguration.yaml         | 1 -
 operator/cmd/mesh/manifest-generate_test.go               | 2 --
 releasenotes/notes/37737.yaml                             | 8 ++++++++
 10 files changed, 8 insertions(+), 13 deletions(-)
 create mode 100644 releasenotes/notes/37737.yaml

diff --git a/manifests/charts/default/templates/mutatingwebhook.yaml b/manifests/charts/default/templates/mutatingwebhook.yaml
index 54dc20cae0..7dc189da13 100644
--- a/manifests/charts/default/templates/mutatingwebhook.yaml
+++ b/manifests/charts/default/templates/mutatingwebhook.yaml
@@ -12,7 +12,6 @@
       namespace: {{ .Release.Namespace }}
       path: "/inject"
     {{- end }}
-    caBundle: ""
   sideEffects: None
   rules:
     - operations: [ "CREATE" ]
diff --git a/manifests/charts/default/templates/validatingwebhook.yaml b/manifests/charts/default/templates/validatingwebhook.yaml
index 8d9fcc05d1..94c4416b53 100644
--- a/manifests/charts/default/templates/validatingwebhook.yaml
+++ b/manifests/charts/default/templates/validatingwebhook.yaml
@@ -21,7 +21,6 @@ webhooks:
         namespace: {{ .Values.global.istioNamespace }}
         path: "/validate"
       {{- end }}
-      caBundle: "" # patched at runtime when the webhook is ready.
     rules:
       - operations:
           - CREATE
diff --git a/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml b/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
index cb78c74cf0..b635bfaaed 100644
--- a/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
+++ b/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml
@@ -2654,7 +2654,6 @@ webhooks:
       namespace: istio-system
       path: "/inject"
       port: 443
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
@@ -2684,7 +2683,6 @@ webhooks:
       namespace: istio-system
       path: "/inject"
       port: 443
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
@@ -2716,7 +2714,6 @@ webhooks:
       namespace: istio-system
       path: "/inject"
       port: 443
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
@@ -2744,7 +2741,6 @@ webhooks:
       namespace: istio-system
       path: "/inject"
       port: 443
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
diff --git a/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml b/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml
index dcb84dde3e..2b896b8e0b 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml
@@ -13,7 +13,6 @@ a unique prefix to each. */}}
       path: "{{ .Values.istiodRemote.injectionPath }}"
       port: 443
     {{- end }}
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
diff --git a/manifests/charts/istio-control/istio-discovery/templates/revision-tags.yaml b/manifests/charts/istio-control/istio-discovery/templates/revision-tags.yaml
index 2ec985f04c..e30e182a19 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/revision-tags.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/revision-tags.yaml
@@ -12,7 +12,6 @@
       namespace: {{ .Release.Namespace }}
       path: "{{ .Values.istiodRemote.injectionPath }}"
     {{- end }}
-    caBundle: ""
   sideEffects: None
   rules:
     - operations: [ "CREATE" ]
diff --git a/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml b/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml
index 15102a1745..72d3adf6d4 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/validatingwebhookconfiguration.yaml
@@ -22,7 +22,6 @@ webhooks:
         namespace: {{ .Values.global.istioNamespace }}
         path: "/validate"
       {{- end }}
-      caBundle: "" # patched at runtime when the webhook is ready.
     rules:
       - operations:
           - CREATE
diff --git a/manifests/charts/istiod-remote/templates/mutatingwebhook.yaml b/manifests/charts/istiod-remote/templates/mutatingwebhook.yaml
index dcb84dde3e..2b896b8e0b 100644
--- a/manifests/charts/istiod-remote/templates/mutatingwebhook.yaml
+++ b/manifests/charts/istiod-remote/templates/mutatingwebhook.yaml
@@ -13,7 +13,6 @@ a unique prefix to each. */}}
       path: "{{ .Values.istiodRemote.injectionPath }}"
       port: 443
     {{- end }}
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
diff --git a/manifests/charts/istiod-remote/templates/validatingwebhookconfiguration.yaml b/manifests/charts/istiod-remote/templates/validatingwebhookconfiguration.yaml
index 8e87383fae..637aba9053 100644
--- a/manifests/charts/istiod-remote/templates/validatingwebhookconfiguration.yaml
+++ b/manifests/charts/istiod-remote/templates/validatingwebhookconfiguration.yaml
@@ -23,7 +23,6 @@ webhooks:
         namespace: {{ .Values.global.istioNamespace }}
         path: "/validate"
       {{- end }}
-      caBundle: "" # patched at runtime when the webhook is ready.
     rules:
       - operations:
           - CREATE
diff --git a/operator/cmd/mesh/manifest-generate_test.go b/operator/cmd/mesh/manifest-generate_test.go
index 3f977089b7..fb36c46bb1 100644
--- a/operator/cmd/mesh/manifest-generate_test.go
+++ b/operator/cmd/mesh/manifest-generate_test.go
@@ -850,7 +850,6 @@ func mergeWebhooks(whs ...[]v1.MutatingWebhook) []v1.MutatingWebhook {
       name: istiod
       namespace: istio-system
       path: "/inject"
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
@@ -883,7 +882,6 @@ func mergeWebhooks(whs ...[]v1.MutatingWebhook) []v1.MutatingWebhook {
       name: istiod-canary
       namespace: istio-system
       path: "/inject"
-    caBundle: ""
   sideEffects: None
   rules:
   - operations: [ "CREATE" ]
diff --git a/releasenotes/notes/37737.yaml b/releasenotes/notes/37737.yaml
new file mode 100644
index 0000000000..61a8eeb759
--- /dev/null
+++ b/releasenotes/notes/37737.yaml
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: installation
+issue:
+  - 33052
+releaseNotes:
+  - |
+    **Fixed** Removing caBundle default value from Chart to allow a GitOps approach
-- 
2.35.3

